File: views/src/main/java/ru/twozeros/todo/views/mapper/TaskMapper.java
package ru.twozeros.todo.views.mapper;

import ru.twozeros.todo.service.model.Task;
import ru.twozeros.todo.views.model.NewTaskDto;
import ru.twozeros.todo.views.model.TaskDto;

import java.util.Date;

public class TaskMapper {

    public static Task toService(TaskDto task) {
        return Task.builder()
                .id(task.getId())
                .name(task.getName())
                .isComplete(task.getIsCompleted())
                .created(new Date())
                .build();
    }

    public static Task toService(NewTaskDto task) {
        return Task.builder()
                .name(task.getName())
                .created(new Date())
                .isComplete(task.getIsCompleted())
                .build();
    }
}

File: views/src/main/java/ru/twozeros/todo/views/mapper/v2/TaskMapper.kt
package ru.twozeros.todo.views.mapper.v2

import ru.twozeros.todo.service.model.Task
import ru.twozeros.todo.views.model.NewTaskDto
import java.util.*

fun NewTaskDto.toService(): Task {
    return Task().also { it ->
        it.name = this.name
        it.complete = this.completed
        it.created = Date()
    }
}